.github/workflows/quality.yml

```diff
@@ -23,4 +23,20 @@ jobs:
           pip install pre-commit
       - name: Run pre-commit
         run: pre-commit run --all-files
-  
\ No newline at end of file
+
+
+  mypy:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Setup Python
+        uses: actions/setup-python@v4
+        with:
+          python-version: "3.12"
+      - name: Checkout
+        uses: actions/checkout@v3
+      - name: Install mypy
+        run: pip install mypy
+      - name: Run mypy
+        uses: sasanquaneuf/mypy-github-action@releases/v1
+        with:
+          checkName: 'mypy'   # NOTE: this needs to be the same as the job name
```

Comment:
should i add argument --check-untyped-defs

Reply:
Not sure, actually! I don't have a strong opinion about this one.

Great detail question, we can discuss and decide in the call.